src/effects/Echo.cpp

```diff
@@ -108,23 +148,25 @@ bool EffectEcho::ProcessInitialize(
       history.reinit(histLen, true);
    }
    catch ( const std::bad_alloc& ) {
-      Effect::MessageBox( XO("Requested value exceeds memory capacity.") );
+      mProcessor.MessageBox( XO("Requested value exceeds memory capacity.") );
       return false;
    }
 
    return history != NULL;
 }
 
-bool EffectEcho::ProcessFinalize()
+bool EffectEcho::Instance::ProcessFinalize()
 {
-   history.reset();
    return true;
 }
 
-size_t EffectEcho::ProcessBlock(EffectSettings &,
+size_t EffectEcho::Instance::ProcessBlock(EffectSettings& settings,
    const float *const *inBlock, float *const *outBlock, size_t blockLen)
 {
-   auto& echoSettings = mSettings;
+   // temporary - in the final step this will be replaced by
+   // auto& echoSettings = GetSettings(settings);
+   //
+   auto& echoSettings = static_cast<const EffectEcho&>(mProcessor).mSettings;
```

Comment:
Ok just that the argument name settings isn’t needed at this step

Reply:
ok, removed